File: frontend/src/app/models/entities/Bill.ts
import { Reparation } from './Reparation';
import { User } from './User';

export class Bill {
    id: number = 0;
    reparationId: number = 0;
    userId: number = 0;
    date: Date = new Date();
    total: number = 0;
}

File: backend/SmartService/Domain/Entities/Bill.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace SmartService.Domain.Entities
{
    public class Bill
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Reparation")]
        public int ReparationId { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        [Column(TypeName = "decimal(10, 2)")]
        public decimal Total { get; set; }

        public User User { get; set; }
        public Reparation Reparation { get; set; }
    }
}
